code_generator/GdkCodeGenerator/Templates/UnityGameObjectComponentDispatcherGenerator.tt

```diff
@@ -85,23 +101,23 @@ namespace <#= qualifiedNamespace #>
 <# } #>
             }
 
-            public override void InvokeOnEventUserCallbacks(GameObjectDispatcherSystem gameObjectDispatcherSystem)
+            public override void InvokeOnEventUserCallbacks(Dictionary<int, ReaderWriterStore> readerWriterStores)
             {
 <# for (var j = 0; j < eventDetailsList.Count; j++) { #>
                 {
                     var entities = EventsReceivedComponentGroups[<#= j #>].GetEntityArray();
                     var eventLists = EventsReceivedComponentGroups[<#= j #>].GetComponentArray<EventsReceived<<#= eventDetailsList[j].EventName #>Event>>();
                     for (var i = 0; i < entities.Length; i++)
                     {
-                        var manager = gameObjectDispatcherSystem.GetSpatialOSBehaviourManager(entities[i].Index);
-                        if (!manager.TryGetReadersWriters(<#= unityComponentDefinition.Id #>, out var readers))
+                        var readerWriterStore = readerWriterStores[entities[i].Index];
+                        if (!readerWriterStore.TryGetReaderWritersForComponent(<#= unityComponentDefinition.Id #>, out var readers))
```

Comment:
Should the `TryGetReaderWriters` be doing the casting at this point, instead of the `OfType` / `Cast` below?

Reply:
No, that part of the code doesn't know about the specific types.

Reply:
Why not?

Reply:
If we wanted that, the store would have to allocate a new HashSet<CorrectType> right? As it is, it just returns the internal collection it has already allocated

Reply:
I see what you mean... well, it depends on whether you would need to use `.OfType` or `.Cast` every time you use this method. Then the cost can still be simplified to this method call.

I can think of quite an elaborate way of resolving the "storing of the right type value" but I'm unsure if it'd be worth it so I'll spare the details :D